### PR TITLE
Implement stop/start notebook methods

### PR DESCRIPTION
**Pull request by Ladicle**

### CODE REVIEW

File: server/internal/server/notebooks.go [hunk 1]
```diff
@@ -149,6 +149,84 @@ func (s *S) GetNotebook(ctx context.Context, req *v1.GetNotebookRequest) (*v1.No
...
+		store.NotebookStateStopped:
+		if err := s.store.UpdateNotebookState(nb.NotebookID, nb.Version, store.NotebookStateQueued); err != nil {
+			return nil, status.Errorf(codes.Internal, "update notebook state: %s", err)
+		}
+		nb.State = store.NotebookStateQueued
+	case store.NotebookStateQueued,
+		store.NotebookStateRunning,
+		store.NotebookStateFailed:
+	default:
+		return nil, status.Errorf(codes.Internal, "unexpected notebook state: %s", nb.State)
```
Comment (kkaneda) [inline server/internal/server/notebooks.go:220]: How about `codes.FailedPrecondition`?

File: server/internal/server/notebooks.go [hunk 2]
```diff
@@ -149,6 +149,84 @@ func (s *S) GetNotebook(ctx context.Context, req *v1.GetNotebookRequest) (*v1.No
...
+		store.NotebookStateRunning:
+		if err := s.store.UpdateNotebookState(nb.NotebookID, nb.Version, store.NotebookStateStopping); err != nil {
+			return nil, status.Errorf(codes.Internal, "update notebook state: %s", err)
+		}
+		nb.State = store.NotebookStateStopping
+	case store.NotebookStateFailed,
+		store.NotebookStateStopping,
+		store.NotebookStateStopped:
+	default:
+		return nil, status.Errorf(codes.Internal, "unexpected notebook state: %s", nb.State)
```
Comment (kkaneda) [inline server/internal/server/notebooks.go:181]: How about `codes.FailedPrecondition`?

We could also gracefully handle (i.e., do nothing) for `Stopping` and `Stopped`.

File: dispatcher/internal/dispatcher/notebook.go [hunk 3]
```diff
@@ -108,6 +111,7 @@ func (n *NotebookManager) createNotebook(ctx context.Context, nb *store.Notebook
 			managedAnnotationKey:    "true",
 			notebookIDAnnotationKey: nb.NotebookID}).
 		WithSpec(appsv1apply.DeploymentSpec().
+			WithReplicas(1).
```
Comment (kkaneda) [inline dispatcher/internal/dispatcher/notebook.go:114]: Just wanted to check my understanding. When a stopped notebook is restarted, we reach here and apply again. This scales up the deployment.

**Review by kkaneda — APPROVED** [review]
LGTM!

File: dispatcher/internal/dispatcher/notebook.go (hunk 3 above)
Comment (Ladicle) [reply dispatcher/internal/dispatcher/notebook.go:114]: Yes, that's correct.

**Pull request merged**